### PR TITLE
Change execution price

### PR DESCRIPTION
**Pull request by raymogg**
# Motivation
Previously we were ensuring that the price of two orders was exactly the same for execution to happen. This however isn't the case. Orders should be executed if their prices cross. A long order price should be greater than or equal to the price of the short order.

# Changes
- adds a `getExecutionPrice` function
- use the execution price in perp
- alter the `canMatch` check to not enforce strict equals
- add new test

### CODE REVIEW

File: contracts/lib/LibPerpetuals.sol [hunk 1]
```diff
@@ -113,4 +114,16 @@ library Perpetuals {
             notFilled &&
             createdBefore;
     }
+
+    function getExecutionPrice(
+        Order calldata a,
+        Order calldata b
+    ) public pure returns (uint) {
+        bool aIsFirst = a.created <= b.created;
```
Comment (sporejack) [inline contracts/lib/LibPerpetuals.sol:126]: ...ternary?

**Review by sporejack — APPROVED** [review]

**Review by CalabashSquash — APPROVED** [review]
:madagascar:

File: contracts/lib/LibPerpetuals.sol (hunk 1 above)
Comment (raymogg) [reply contracts/lib/LibPerpetuals.sol:126]: @sporejack ternary seems a bit overkill here no? A ternary where you go `isThingTrue ? true : false` is the same as returning `isThingTrue`

**Review by kumar-ish — APPROVED** [review]
:fish:

**Pull request merged**